### DISCUSSION TITLE
[Analytical Table]: Ignore selection for disabled rows when selectSubRows is true

### DISCUSSION THREAD

**Discussion by shehabshalan1** [UI5/webcomponents-react#5676] [Q&A] [answered]
Hi, 
We are currently using the Analytical Table component to render a tree table. A parent or a child row can be disabled based on certain conditions. For this we utilize `useRowDisableSelection` hook. We also use `useIndeterminateRowSelection` with `reactTableOptions={{ selectSubRows: true }}`. 
I noticed that when selecting a parent, the disabled rows in the children are also selected which should not be the case. Ideally a row can be selected if it is not disabled.

I made a stackblitz [here](https://stackblitz.com/edit/github-ahioth-9su23c) to show this behavior and video below as well. 

https://github.com/SAP/ui5-webcomponents-react/assets/130651515/d3490664-8c32-4cea-85dc-80aa90040aae

A workaround I tried is to toggle the disabled row selection off again when parent row is selected but this workaround only works if the parent is already expanded.

**Comment by koeppel** [comment]
Hi @shehabshalan1,

I've updated the sample and created an issue on the repository, I assume that the colleagues will take over from there :)
https://github.com/SAP/ui5-webcomponents-react/issues/6965

Also: the `useRowDisableSelection` is deprecated now in a later release, I assume that this issue will be marked as "works as intended" as the disabled property shouldn't be used anymore..

BR,
Janik

**Reply by koeppel** [reply, marked as answer]
I received an answer to the issue and the design guidelines state what I also mentioned beforehand: disabled rows are no longer a good feature thats why `useRowDisableSelection` is now deprecated. I assume the wanted behavior is simply not feasable.

**Reply by shehabshalan1** [reply]
Hi @koeppel, 

Thanks for the update.
